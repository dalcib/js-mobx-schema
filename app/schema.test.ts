import ClassDecoratorExample, { classSchema } from './example'
import Model from './model'
import { autorun, whyRun, isObservable } from 'mobx'

//const classDecoratorExample = new ClassDecoratorExample()
const example = new ClassDecoratorExample()
//const model = new Model(ClassDecoratorExample)

describe('non-static decorator', () => {
  it('static', () => {
    expect(ClassDecoratorExample.schema).toBeDefined()
  })
})

describe('class', () => {
  it('instanciate', () => {
    expect(example).toMatchSnapshot()
  })
})

describe('schema', () => {
  it('snapshot', () => {
    expect(ClassDecoratorExample.schema).toMatchSnapshot()
  })
})

describe('schema', () => {
  it('generated', () => {
    expect(ClassDecoratorExample.schema).toEqual(classSchema)
  })
})
