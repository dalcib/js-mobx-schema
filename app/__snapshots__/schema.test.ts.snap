// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class instanciate 1`] = `
ClassDecoratorExample {
  "age": 16,
  "date": "2017-08-08",
  "point": false,
}
`;

exports[`schema snapshot 1`] = `
Object {
  "$schema": "http://json-schema.org/schema#",
  "description": "teste",
  "id": "ClassDecoratorExample",
  "properties": Object {
    "age": Object {
      "maximum": 23,
      "type": "number",
    },
    "ages": Object {
      "type": "number",
    },
    "cars": Object {
      "minimum": 4,
      "type": "integer",
    },
    "date": Object {
      "format": "date",
      "type": "string",
    },
    "lastName": Object {
      "minLength": 5,
      "type": "string",
    },
    "list": Object {
      "type": "array",
    },
    "name": Object {
      "default": "fghkkk",
      "title": "Name",
      "type": "string",
    },
  },
  "required": Array [
    "lastName",
  ],
  "title": "ClassDecoratorExample",
  "type": "object",
}
`;
